### DISCUSSION TITLE
[Elsa 2.5.0] Retry a custom activity

### DISCUSSION THREAD

**Discussion by stanroxana** [elsa-workflows/elsa-core#2774] [Q&A] [answered]
We have a scenario in our workflow where we want to retry a process, using a custom activity.
My first thought was to _use the workflow's variables_ to save the number of current retries, but it doesn't seem to work as I was expecting. The variables won't persist if the activity is triggered again.

**The workflow that I am trying to configure looks like this**:
![image](https://user-images.githubusercontent.com/11271010/155512066-8001cf50-8c6f-47ed-99da-ce22172cf93f.png)

So I have a maximum number of retries set in the Retry activity. The idea is that after the workflow passes by the Retry activity, it should go back to the OCR activity until it is successful, iterating for maximum the number of retries set in the Retry activity.

My question is if our use case would be possible right now in Elsa, or if I should look for a workaround.

Thank you so much for your help! 👍

**Comment by sfmskywalker** [comment, marked as answer]
Hi @stanroxana 

That should certainly be possible! Your Retry activity could maintain its current retry count using activity state (no need to use WF variables, which are global to the entire workflow).

For an example, take a look at the [For](https://github.com/elsa-workflows/elsa-core/blob/master/src/core/Elsa.Core/Activities/ControlFlow/For/For.cs#L32) activity, which also needs to maintain internal state like current value.

The trick is to use the `GetState` and `SetState` methods, which will store custom activity data alongside the activity's state.

**Reply by stanroxana** [reply]
Hi Sipke,

Thank you for the hint. I was actually thinking that the For activity has to have the same mechanism afterall. But I didn't pay attention to the exact part that you mentioned, which was the key: the GetState and SetState methods.
Will definitely go back and give it a try.
